phpunit/class-wp-theme-json-test.php

```diff
@@ -5771,83 +5771,182 @@ public function test_resolve_variables() {
 			),
-			'id selector'              => array(
+			'id selector'                     => array(
 				'selector' => '#wp-block',
 				'expected' => '#wp-block.is-style-custom',
 			),
-			'element tag selector'     => array(
+			'element tag selector'            => array(
 				'selector' => 'p',
 				'expected' => 'p.is-style-custom',
 			),
-			'attribute selector'       => array(
+			'attribute selector'              => array(
 				'selector' => '[style*="color"]',
 				'expected' => '[style*="color"].is-style-custom',
 			),
-			'descendant selector'      => array(
+			'descendant selector'             => array(
 				'selector' => '.wp-block .inner',
 				'expected' => '.wp-block.is-style-custom .inner',
 			),
-			'comma separated selector' => array(
+			'comma separated selector'        => array(
 				'selector' => '.wp-block .inner, .wp-block .alternative',
 				'expected' => '.wp-block.is-style-custom .inner, .wp-block.is-style-custom .alternative',
 			),
-			'pseudo selector'          => array(
+			'pseudo selector'                 => array(
 				'selector' => 'div:first-child',
 				'expected' => 'div.is-style-custom:first-child',
 			),
-			':is selector'             => array(
+			':is selector'                    => array(
 				'selector' => '.wp-block:is(.outer .inner:first-child)',
 				'expected' => '.wp-block.is-style-custom:is(.outer .inner:first-child)',
 			),
-			':not selector'            => array(
+			':not selector'                   => array(
 				'selector' => '.wp-block:not(.outer .inner:first-child)',
 				'expected' => '.wp-block.is-style-custom:not(.outer .inner:first-child)',
 			),
-			':has selector'            => array(
+			':has selector'                   => array(
 				'selector' => '.wp-block:has(.outer .inner:first-child)',
 				'expected' => '.wp-block.is-style-custom:has(.outer .inner:first-child)',
 			),
-			':where selector'          => array(
+			':where selector'                 => array(
 				'selector' => '.wp-block:where(.outer .inner:first-child)',
 				'expected' => '.wp-block.is-style-custom:where(.outer .inner:first-child)',
 			),
-			'wrapping :where selector' => array(
+			'wrapping :where selector'        => array(
 				'selector' => ':where(.outer .inner:first-child)',
 				'expected' => ':where(.outer.is-style-custom .inner:first-child)',
 			),
-			'complex'                  => array(
+			'complex'                         => array(
 				'selector' => '.wp:where(.something):is(.test:not(.nothing p)):has(div[style]) .content, .wp:where(.nothing):not(.test:is(.something div)):has(span[style]) .inner',
 				'expected' => '.wp.is-style-custom:where(.something):is(.test:not(.nothing p)):has(div[style]) .content, .wp.is-style-custom:where(.nothing):not(.test:is(.something div)):has(span[style]) .inner',
 			),
+			'child combinator selector'       => array(
+				'selector' => '.wp-block-list > li',
+				'expected' => '.wp-block-list.is-style-custom > li',
```

Comment:
This test reflects what's being rendered, e.g., `:root :where(.wp-block-list.is-style-my-theme-tag--1 > li)`

Isn't the expected value, for the list-item variation selectors to apply correctly, `:root :where(.wp-block-list > li.is-style-my-theme-tag--1`

I tried adding `"blockElement": "li"` to packages/block-library/src/list-item/block.json and it seemed to work on the front end at least.

Reply:
This test illustrates a block style variation for a block that might also target an inner element for some block support features, e.g. throw border styles onto the inner `li` elements individually. 

So this test is really covering a block style variation on the List block, not the ListITem block.

The use case for the ListItem block is below:

https://github.com/WordPress/gutenberg/pull/73719/files#diff-88c997ea8e8ba51f2fa201327e0f1faccd4e892c2fafe33249f72cfd068c72fcR5868-R5871

```php
			'list item basic'                 => array(
				'selector'      => '.wp-block-list > li',
				'expected'      => '.wp-block-list > li.is-style-custom',
				'block_element' => 'li',
			),
```

Or, further below that again if the `blockElement` was explicitly `> li`

Reply:
Sorry for wasting your time, I completely missed that 🤦🏻 Should be reading more closely.